src/selectors/columnWidths.js

```diff
@@ -149,10 +162,23 @@ function groupColumns(columnProps) {
     container.push(columnProp);
   });
 
+  // group the scrollable column groups together and also provide
+  // index mapping (and inverse mapping) from the column group to it's order in the view
+  forEach(columnGroupProps, (columnProp, index) => {
```

Comment:
so this is where it got tricky. I'll explain this over an example.

FDT allows us to specify column/column groups in any order. The order determines how these are displayed in the viewport, but this isn't exactly true.
If any of these are `fixed` or `fixedRight`, then they obviously starts at the start/end and move towards the middle.
See examples [Fixed Left columns](http://ld-desis.dev.bb.schrodinger.com:8090/example-object-data.html) and [Fixed RightColumns](http://ld-desis.dev.bb.schrodinger.com:8090/fixed-right-columns.html) to get an idea.

So if the column groups are specified like 
```
F, F, S, S, FR, S, F, S (col)
0, 1, 2, 3, 4,  5, 6, 7 (index)
```
where F is fixed, S is scrollable, and FR is fixed right,
then the user will see it as
```
F, F, F, S, S, S, S, FR (col)
0, 1, 6, 2, 3, 5, 7, 4  (index)
```

so now even if we get the first and last column group in the viewport as 2 and 7, we can't iterate over, since `3`, `4`, `6` may not be the correct ones. (Here the correct answer will be `2, 3, 5, 7`)

So what we do is maintain a map called `columnGroupIndex` that gives the index ordering *just* for the scrollable column groups
So something like
```
F, F, F, S, S, S, S, FR (col)
0, 1, 6, 2, 3, 5, 7, 4  (index)
X, X, X, 0, 1, 2, 3, X  (columnGroupIndex)
```

now we'll use this map to get the range of columns, which is `0, 1, 2, 3`
Notice we don't need to do this for fixed/fixedRight columns since they're not virtualized.

So now that we got the range, we need to get the actual scrollable column group. So we need another map called `scrollableColumnGroups` to do the inverse mapping
so
```
F, F, S, S, FR, S, F, S (col)
0, 1, 2, 3, 4,  5, 6, 7 (index)
X, X, 0, 1, X, 2, X, 3  (columnGroupIndex)
2, 3, 5, 6, X, X, X, X  (scrollableColumnGroups)
```

and tadaaa! we got the actual column group props